fvm/evm/handler/handler.go

```diff
@@ -246,19 +234,44 @@ func (h *ContractHandler) batchRun(rlpEncodedTxs [][]byte, coinbase types.Addres
 		h.tracer.Collect(r.TxHash)
 	}
 
-	err = h.emitEvent(types.NewBlockEvent(bp))
+	// update the block proposal
+	err = h.blockStore.UpdateBlockProposal(bp)
 	if err != nil {
 		return nil, err
 	}
 
-	err = h.blockStore.CommitBlockProposal()
+	// TODO: don't commit right away.
+	err = h.CommitBlockProposal()
 	if err != nil {
 		return nil, err
 	}
 
 	return res, nil
 }
 
+// CommitBlockProposal commits the block proposal and add a new block to the EVM blockchain
+func (h *ContractHandler) CommitBlockProposal() error {
```

Comment:
Later we can expose this to the system chunk transaction instead of calling it on every interaction.